service/storage_service.cc

```diff
@@ -5422,23 +5497,46 @@ future<node_ops_cmd_response> storage_service::node_ops_cmd_handler(gms::inet_ad
                 slogger.warn("{}", msg);
                 throw std::runtime_error(msg);
             }
-            mutate_token_metadata([coordinator, &req, this] (mutable_token_metadata_ptr tmptr) mutable {
+            mutate_token_metadata([coordinator, coordinator_host_id, &req, this] (mutable_token_metadata_ptr tmptr) mutable {
                 for (auto& x: req.replace_nodes) {
                     auto existing_node = x.first;
                     auto replacing_node = x.second;
-                    slogger.info("replace[{}]: Added replacing_node={} to replace existing_node={}, coordinator={}", req.ops_uuid, replacing_node, existing_node, coordinator);
-                    tmptr->update_topology(replacing_node, get_dc_rack_for(replacing_node), locator::node::state::replacing);
-                    tmptr->add_replacing_endpoint(existing_node, replacing_node);
+                    const auto existing_node_id = tmptr->get_host_id(existing_node);
+                    const auto replacing_node_id = coordinator_host_id;
```

Comment:
I wonder, isn't it possible to obtain this from gossiper? I guess in the case of replace-with-same-IP, gossiper might be mapping the IP to the old host ID, but is that indeed so?

Reply:
> gossiper might be mapping the IP to the old host ID, but is that indeed so?

I believe so. I remember I tried to use `gossiper->get_host_id` and some tests failed. I liked the `coordinator_host_id`-based solution more, since it's simple and direct. Everything involving gossiper is usually hairy and difficult to deal with.